File: javascript/rna-transcription/rna-transcription.js
const dnaNucleotideComplement = {
  G: 'C',
  C: 'G',
  T: 'A',
  A: 'U',
};

const dnaToRnaNucleotide = nucleotide => {
  return dnaNucleotideComplement[nucleotide];
};

export const toRna = dnaStrand => {
  if (!dnaStrand) return '';

  return dnaToRnaNucleotide(dnaStrand[0]) + toRna(dnaStrand.substr(1));
};

File: go/rna-transcription/rna_transcription.go
package strand

import "strings"

var dnaToRna = map[rune]rune{
	'G': 'C',
	'C': 'G',
	'T': 'A',
	'A': 'U',
}

func ToRNA(dna string) string {
	sb := strings.Builder{}

	for _, nucleotide := range dna {
		sb.WriteRune(dnaToRna[nucleotide])
	}

	return sb.String()
}
